### PR TITLE
Add convenience method to the `MockConnection` 

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description

See branch name for motivation, you can now remove components & an entity the same way you add them.

#### Tests

- [x] Updated a test to double check!

#### Documentation

- [x] Changelog?

### CODE REVIEW

File: test-project/Assets/EditmodeTests/Collections/ReferenceProviderTests.cs [hunk 1]
```diff
@@ -35,25 +29,30 @@ public void Disposed_world_cleans_provider()
...
 
                 Assert.AreEqual(0, ReferenceProvider<string>.Count);
             }
         }
+
+        private EntityTemplate GetTemplate()
+        {
+            var template = new EntityTemplate();
+            template.AddComponent(new Position.Snapshot(), "some-worker");
+            template.AddComponent(new Metadata.Snapshot(), "some-worker");
```
Comment (zeroZshadow) [inline test-project/Assets/EditmodeTests/Collections/ReferenceProviderTests.cs:54]: This should have a value to keep the tests the same

File: test-project/Assets/EditmodeTests/Collections/ReferenceProviderTests.cs [hunk 2]
```diff
@@ -35,25 +29,30 @@ public void Disposed_world_cleans_provider()
...
-                });
+                mockWorld
+                    .Step(world =>
+                    {
+                        world.Connection.CreateEntity(entityId, template);
+                    })
+                    .Step(world =>
+                    {
+                        Assert.AreEqual(1, ReferenceProvider<string>.Count);
+                        world.Connection.RemoveEntityAndComponents(entityId, template);
```
Comment (zeroZshadow) [inline test-project/Assets/EditmodeTests/Collections/ReferenceProviderTests.cs:43]: The test previously only removed the component, not the whole entity?

File: test-project/Assets/EditmodeTests/Collections/ReferenceProviderTests.cs (hunk 1 above)
Comment (jamiebrynes7) [reply test-project/Assets/EditmodeTests/Collections/ReferenceProviderTests.cs:54]: Hahaha this actually means we pushed a `null` into the provider

File: test-project/Assets/EditmodeTests/Collections/ReferenceProviderTests.cs (hunk 2 above)
Comment (jamiebrynes7) [reply test-project/Assets/EditmodeTests/Collections/ReferenceProviderTests.cs:43]: Discussed offline - this is what created the unhappiness in the first place (the fact that `RemoveEntity` didn't work originally). So we are good to swap this over

**Pull request merged**